bin/cassandra.in.sh

```diff
@@ -122,13 +122,22 @@ jvmver=`echo "$java_ver_output" | grep '[openjdk|java] version' | awk -F'"' 'NR=
 JVM_VERSION=${jvmver%_*}
 short=$(echo "${jvmver}" | cut -c1-2)
 
-JAVA_VERSION=17
 if [ "$short" = "11" ]  ; then
      JAVA_VERSION=11
 elif [ "$JVM_VERSION" \< "17" ] ; then
     echo "Cassandra 5.0 requires Java 11 or Java 17."
     exit 1;
 fi
+# Allow execution if a supported Java version is used or CASSANDRA_USE_ALL_JDK argument is set
+java_versions_supported=11,17
+supported_version=$(echo "$java_versions_supported" | tr "," '\n' | grep -F -x "$short")
+
+if [ "x$CASSANDRA_USE_ALL_JDK" != "xtrue" ] && [ "x$supported_version" = "x" ] ; then
+    echo "Cassandra 5.0 requires Java 11, Java 17 (or newer LTS)."
```

Comment:
```suggestion
    echo "Cassandra 5.0 requires Java 11 or Java 17."
```


comment applies to all places.

Reply:
- I have added the changes to `redhat/cassandra.in.sh`. 
- `debian/patches/cassandra_in.sh_dirs.diff` does not check for Java version. Should I add?
How do I test these?

Reply:
@michaelsembwever, @smiklosovic,  please correct me if I am wrong, but I think after https://github.com/apache/cassandra/commit/e76b63d08146a3918b8315b9f0a11a04dca66da7, we allow only 11 and 17 with Debian packages? 
Also, regarding testing, I think we need to build the artifacts and then try to run cassandra?

Reply:
Both debian and redhat packages require jdk11 or jdk17 dependencies.

But it's easy to get around package deps, so enforcing it here still applies 👍